### DISCUSSION TITLE
How to use `_known_hosts_real` bash completion function?

### DISCUSSION THREAD

**Discussion by postmodern** [DannyBen/completely#65] [Ideas]
Instead of using the regular `<hostname>` suggestion keyword, I'd like to use Bash's `_known_hosts_real` function, which parses ssh's `known_hosts` file and `avahi-browse -cpr _workstation._tcp` to list commonly accessed host names.

**Comment by DannyBen** [comment]
There are three things that can be used as completions with completely:
- any string
- anything supported by [`compgen -A action`](https://www.gnu.org/software/bash/manual/html_node/Programmable-Completion-Builtins.html)
- any command available in your shell that returns whitespace delimited strings, using the `$(...)` syntax.

As I understand, `_known_hosts_real` is meant to be a private function (judging by the leading `_`, and it does not fit into any of the above categories. While it is available in your shell, it is not available to subshells (`bash -c _known_hosts_real` does not work).

So bottom line, unless you have a function that you can run in your shell, and that returns (outputs) whitespace delimited list of known hosts, I cannot think of a way to use it.

**Comment by DannyBen** [comment]
Another thing - in case you are looking for alternatives.

Using this configuration will show the known hosts from your ssh config:

```yaml
# completely.yaml
mygit:
- $(grep -P "^Host ([^*]+)$" $HOME/.ssh/config | sed 's/Host //')
```

You can test it with `completely test "mygit <search string>"`

Perhaps you can use a similar approach to parse any other file that contains the desired information.
